Flaky E2E-Tests fixen

Einige E2E-Tests sind flaky, was dazu führt das man die Pipeline babysitten muss. Sie schlagen in der Pipeline fehl, sind dann aber bei erneutem Ausführen OK.

**Das Ticket soll [Timeboxed](https://resources.scrumalliance.org/Article/timeboxing-scrum) sein!**

### `check-in.cy.ts`
```
  1) okr check-in
       should have a primary button on every check-in dialog:
     AssertionError: Timed out retrying after 4000ms: Expected to find element: `[data-testId=show-all-checkins]`, but never found it.
      at Context.eval (webpack://frontend/./cypress/support/commands.ts:26:0)
```
Beispiel: https://github.com/puzzle/okr/actions/runs/12930016350/job/36060603408

### ...

### Anforderungen
- Die E2E-Tests funktionieren wieder zuverlässig(er)

### Akzeptanzkriterien
- [ ] Die Tests läuft sowohl lokal als auch auf der pipeline nicht mehr flaky.
- [ ] Die Tests läuft sowohl headless als auch nicht ohne flaky zu sein.
- [ ] Die Tests wurde nach dem Fix >20 Mal hintereinander in der Pipeline ausgeführt, ohne das sie ein einziges Mal fehlgeschlagen sind

--- a/frontend/cypress.config.ts
+++ b/frontend/cypress.config.ts
@@ -10,5 +10,11 @@ export default defineConfig({
   },
   env: {
     LOGIN_URL: 'http://localhost:8544'
-  }
+  },
+  retries: {
+    // Configure retry attempts for `cypress run`
+    runMode: 2,
+    // Configure retry attempts for `cypress open`
+    openMode: 0,
+  },
 });

